src/common/perf_counters.cc

```diff
@@ -138,7 +138,12 @@ void PerfCountersCollectionImpl::dump_formatted_generic(
   
   if (dump_labeled) {
     std::string prev_key_name;
+    bool array_open = false;
     for (auto l = m_loggers.begin(); l != m_loggers.end(); ++l) {
+      if ((*l)->time_filtered && ((*l)->last_updated + (*l)->time_alive < ceph::coarse_real_clock::now())) {
```

Comment:
all of the writes to `PerfCounters::last_updated` happen under the `PerfCountersCache::m_lock` mutex, but we read it here without the lock

hopefully that's okay? i can't think of a good way to avoid it

Reply:
Wouldn't the same concern apply to the value of the underlying counter that's being changed, for example by `PerfCounters::inc(index, value)` in `PerfCountersCache::inc()`?

Reply:
PerfCounters::inc() is thread-safe because it writes to atomic variables. PerfCountersCache::inc() is thread-safe because it holds a mutex